.github/workflows/xe-experimental-vm-test.yml

```diff
@@ -0,0 +1,35 @@
+name: "[experimental] VM-based integration testing, ping @Xe if it fails"
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - 'main'
+  release:
+    types: [ created ]
+  schedule:
+    - cron: '00 6,18 * * *'
+
+jobs:
+  experimental-linux-vm-test:
+    runs-on: [ self-hosted, linux, vm_integration_test ]
+
+    if: "!contains(github.event.head_commit.message, '[ci skip]')"
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v1
+
+      - name: Download VM Images
+        run: go test ./tstest/integration/vms -run-vm-tests -run Download -timeout 60m
+        env:
+          XDG_CACHE_HOME: "/var/lib/ghrunner/cache"
+
+      - name: Run VM tests
+        run: go test ./tstest/integration/vms -v -run-vm-tests -run Integration
```

Comment:
should be able to drop `--run` here entirely, right? The Download should be a no-op if it already exists on disk?

Reply:
Is there a way to do test-specific timeouts? I have them as separate steps so cache warming has a longer timeout.

Reply:
You already did Download in the previous Step with a large timeout. If it's already on disk, you can run it again and it'll be quick, without needing a large timeout.

Reply:
Oh, true.